lib/db/transactions.go

```diff
@@ -723,15 +728,12 @@ func (t *readWriteTransaction) withAllFolderTruncated(folder []byte, fn func(dev
 			}
 			continue
 		}
-		var f FileInfoTruncated
-		// The iterator function may keep a reference to the unmarshalled
-		// struct, which in turn references the buffer it was unmarshalled
-		// from. dbi.Value() just returns an internal slice that it reuses, so
-		// we need to copy it.
-		err := f.Unmarshal(append([]byte{}, dbi.Value()...))
+
+		intf, err := t.unmarshalTrunc(dbi.Value(), true)
```

Comment:
Isn't the commend above valid (anymore)? `unmarshalTrunc` doesn't do any buffer copying.

Reply:
It wasn't valid at all since we switched to protobuf, which doesn't keep references (or we'd be fucked in so many places...).

This change in general isn't strictly needed right now, but I think all unmarshalling should use the unmarshalTrunc function. (Again, when indirecting version vectors that also affects the truncated version.)